ibm/table_ibm_is_flow_log.go

```diff
@@ -0,0 +1,187 @@
+			{Name: "tags", Type: proto.ColumnType_JSON, Hydrate: getFlowLogTags, Transform: transform.FromValue(), Description: resourceInterfaceDescription("tags")},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listIsFlowLog(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	// Create service connection
+	conn, err := vpcService(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("ibm_is_flow log.listIsFlowLog", "connection_error", err)
+		return nil, err
+	}
+
+	// Retrieve the list of flow log collectors for your account.
+	maxResult := int64(100)
+	start := ""
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	limit := d.QueryContext.Limit
+	if d.QueryContext.Limit != nil {
+		if *limit < maxResult {
+			maxResult = *limit
+		}
+	}
+
+	opts := &vpcv1.ListFlowLogCollectorsOptions{
+		Limit: &maxResult,
+	}
+
+	// Additional filters
+	if d.KeyColumnQuals["name"] != nil {
+		opts.SetName(d.KeyColumnQuals["name"].GetStringValue())
+	}
+
+	for {
+		if start != "" {
+			opts.Start = &start
+		}
+		result, resp, err := conn.ListFlowLogCollectorsWithContext(ctx, opts)
+		if err != nil {
+			plugin.Logger(ctx).Error("ibm_is_flow_log.listIsFlowLog", "query_error", err, "resp", resp)
+			return nil, err
+		}
+		for _, i := range result.FlowLogCollectors {
+			d.StreamListItem(ctx, i)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		start = GetNext(result.Next)
+		if start == "" {
+			break
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getIsFlowLog(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	region := plugin.GetMatrixItem(ctx)["region"].(string)
+
+	// Create service connection
+	conn, err := vpcService(ctx, d, region)
+	if err != nil {
+		plugin.Logger(ctx).Error("ibm_is_flow_log.getIsFlowLog", "connection_error", err)
+		return nil, err
+	}
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// No inputs
+	if id == "" {
+		return nil, nil
+	}
+
+	opts := &vpcv1.GetFlowLogCollectorOptions{
+		ID: &id,
+	}
+
+	result, resp, err := conn.GetFlowLogCollectorWithContext(ctx, opts)
+	if err != nil {
+		plugin.Logger(ctx).Error("ibm_is_flow_log.getIsFlowLog", "query_error", err, "resp", resp)
+		if err.Error() == "Flow log collector not found" {
+			return nil, nil
+		}
+		return nil, err
+	}
+	return *result, nil
+}
+
+func getFlowLogTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	vpc := h.Item.(vpcv1.FlowLogCollector)
+	conn, err := tagService(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("ibm_is_flow_log.getFlowLogTags", "connection_error", err)
+		return nil, err
+	}
+
+	opts := conn.NewListTagsOptions()
+	opts.SetLimit(100)
```

Comment:
Why this limit?

Reply:
100 is the max Limit in [docs](https://cloud.ibm.com/apidocs/vpc/latest?code=go#list-flow-log-collectors)